tests/Unit/Send/ChainTest.php

```diff
@@ -139,4 +138,37 @@ public function testProcessNotProcessed() {
 
 		$this->chain->process($account, $localMessage);
 	}
+
+	public function testProcessDontProcessErrorerdMessages() {
+		$mailAccount = new MailAccount();
+		$mailAccount->setSentMailboxId(1);
+		$mailAccount->setUserId('bob');
+		$account = new Account($mailAccount);
+		$localMessage = new LocalMessage();
+		$localMessage->setId(100);
+		$localMessage->setStatus(LocalMessage::STATUS_ERROR);
+		$expected = new LocalMessage();
+		$expected->setStatus(LocalMessage::STATUS_IMAP_SENT_MAILBOX_FAIL);
+		$expected->setId(100);
+
+		$this->sentMailboxHandler->expects(self::any())
+			->method('setNext')
+			->withConsecutive(
+				[$this->antiAbuseHandler],
+				[$this->sentMailboxHandler],
+				[$this->copySentMessageHandler],
+				[$this->flagRepliedMessageHandler],
+			);
+		$this->sentMailboxHandler->expects(self::never())
+			->method('process');
+		$this->attachmentService->expects(self::never())
+			->method('deleteLocalMessageAttachments');
+		$this->localMessageMapper->expects(self::never())
+			->method('deleteWithRecipients');
+		$this->localMessageMapper->expects(self::never())
+			->method('update');
```

Comment:
you are testing the behavior of sentMailboxHandler here. SentMailboxHandler is a stub and calling process will never call any other methods. If you want to test this path, do it in the unit test for the sent mailbox handler.